assets/main.js.liquid

```diff
@@ -81,33 +81,189 @@ Site.Header = {
+      // window is wider than minimum width (720px) for fixed header layout
+
+      if (!_this.loadedImages) {
+        // bind imagesLoaded
+        _this.$productGallery.imagesLoaded( function() {
+          // remove min-height for product gallery holder
+          _this.$productGallery.removeClass('min-height');
+          // size image holder to fit image size
+          _this.sizeImageHolder();
+          // get dimensions for toggleFixed()
+          _this.getFixedHeaderDimensions();
+          // toggle fixed header
+          _this.toggleFixed();
+          // init product image zooming
+          _this.bindZoom();
+          // images are loaded
+          _this.loadedImages = true;
+        });
+      } else {
+        // on window resize
+        // if images are already loaded
+        // do the layout again
+        _this.sizeImageHolder();
+        _this.getFixedHeaderDimensions();
+        _this.toggleFixed();
+        _this.bindZoom();
+      }
+
+      $(window).on('scroll', function() {
+        // toggle fixed header on scroll
+        _this.toggleFixed();
+      });
+
+    } else {
+      // viewport is mobile width
+
+      // remove min-height for product gallery holder
+      _this.$productGallery.removeClass('min-height');
+      // reset image holder dimensions
+      $('.product-gallery-image-holder').width('auto').height('auto');
+      // unbind zooming
+      _this.destroyZoom();
+      // unbind toggleFixed() on scroll
+      $(window).off('scroll');
+    }
+  },
+
+  getFixedHeaderDimensions: function() {
+    var _this = this;
+
+    // get gallery and header dimensions for
+    // toggling / positioning fixed header
 
-    $('#product-gallery-row').width(galleryWidth);
+    _this.galleryHeight = _this.$productGallery.outerHeight(true);
+    _this.galleryOffset = _this.$productGallery.offset().top;
+    _this.headerHeight = _this.$productHeader.outerHeight(true);
+  },
+
+  toggleFixed: function() {
+    var _this = this;
+
+    var windowScroll = $(window).scrollTop();
+
+    var containerWidth = _this.$container.width();
+
+    if ((windowScroll + _this.galleryOffset + _this.headerHeight) <= (_this.galleryHeight + _this.galleryOffset)) {
+      // fixed header isn't at the bottom of the product gallery holder yet
+
+      // keep header fixed
+      _this.$productHeader.removeClass('bottom');
+      // position fixed header
+      _this.$productHeader.css('margin-right', ((_this.windowWidth - containerWidth) / 2) + 'px');
+    } else {
+      // fixed header is at the bottom of the product gallery
+
+      // position header absolute
+      _this.$productHeader.addClass('bottom');
+      // reset fixed header positioning
+      _this.$productHeader.css('margin-right', 0);
+    }
   },
 
   pickRelated: function() {
-    $('.related-products-item').pick(4);
+    var _this = this;
 
+    // show some related product items
+    $('.related-products-item').pick(_this.numberRelatedToPick);
```

Comment:
Thats actually a pretty sweet jQ plugin

Reply:
shopify made it cause liquid is so limited

Reply:
lmao